### PR TITLE
Virtual core dependency

### LINKED ISSUE #28: Separate Common Logic From Alexa and Google

### PR DESCRIPTION
**Pull request by jperata**
Closes #28

### CODE REVIEW

File: src/BuiltinUtterances.ts [hunk 1]
```diff
@@ -1,34 +0,0 @@
-// Holder class for utterances for builtin intents
-// This could have been done as just a JSON file, but then requires build tool to include it in the lib dir
-export class BuiltinUtterances {
```
Comment (jkelvie) [inline src/BuiltinUtterances.ts]: I meant for this to stay in virtual alexa - google will have its own builtinutterances that must be loaded when using it.

File: src/BuiltinSlotTypes.ts [hunk 2]
```diff
@@ -1,70 +0,0 @@
-import {ISlotValue, SlotMatch, SlotType} from "./SlotTypes";
```
Comment (jkelvie) [inline src/BuiltinSlotTypes.ts]: These BuiltinSlotTypes are Amazon-specific and should stay here.

**Comment by jperata** [comment]
@jkelvie Added back Amazon Builtin Utteranes and Slots.

**Comment by jperata** [comment]
Add a fix to typedocs generation and regenerated the docs.

**Pull request merged**